kimchi/src/circuits/constraints.rs

```diff
@@ -583,9 +587,11 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
         //
         // Lookup
         // ------
-
         let lookup_constraint_system =
-            LookupConstraintSystem::create(&gates, lookup_tables, &domain).ok()?;
+            match LookupConstraintSystem::create(&gates, lookup_tables, &domain) {
+                Ok(ok) => ok,
+                Err(_) => return Err(ProverError::SetupError),
+            };
```

Comment:
can you rewrite this as a one-liner as well as preserve the error when you bubble it up? Something like

```rust
let lookup_constraint_sytem = LookupConstraintSystem::create(&gates, lookup_tables, &domain).map_err(|e| ProverError::SetupError(e))?;
```
```